### DISCUSSION TITLE
Why does clickhouse sink connector terminate with an error?

### DISCUSSION THREAD

**Discussion by zTranquilizer** [ClickHouse/clickhouse-kafka-connect#460] [Q&A] [answered]
Hey, everybody!
Started kafka connect with clickhouse sink connector.
When sending a message to the queue - clickhouse sink connector starts processing the message and then terminates with an error.
Hey, everybody!
Started kafka connect with clickhouse sink connector.
When sending a message to the queue - clickhouse sink connector starts processing the message and then terminates with an error.
Why does clickhouse sink connector terminate with an error?

`
java.lang.VerifyError: Bad return type
Exception Details:
Location:    
io/confluent/protobuf/MetaProto$Meta.internalGetMapFieldReflection(I)Lcom/google/protobuf/MapFieldReflectionAccessor; @24: areturn
Reason:
Type 'com/google/protobuf/MapField' (current frame, stack[0]) is not assignable to 'com/google/protobuf/MapFieldReflectionAccessor' (from method signature)
Current Frame:
bci: @24
flags: { }
locals: { 'io/confluent/protobuf/MetaProto$Meta', integer }
stack: { 'com/google/protobuf/MapField' }
Bytecode:
{{hidden}}
Stackmap Table:
same_frame(@20)
same_frame(@25)
io.confluent.protobuf.MetaProto.<clinit>(MetaProto.java:1112)
io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema.<clinit>(ProtobufSchema.java:246)
io.confluent.kafka.schemaregistry.protobuf.ProtobufSchemaProvider.parseSchemaOrElseThrow(ProtobufSchemaProvider.java:38)\n\tat io.confluent.kafka.schemaregistry.SchemaProvider.parseSchema(SchemaProvider.java:75)\n\tat io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.parseSchema(CachedSchemaRegistryClient.java:301)\n\tat io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaByIdFromRegistry(CachedSchemaRegistryClient.java:347)\n\tat io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaBySubjectAndId(CachedSchemaRegistryClient.java:472)\n\tat io.confluent.kafka.serializers.protobuf.AbstractKafkaProtobufDeserializer.deserialize(AbstractKafkaProtobufDeserializer.java:138)\n\tat io.confluent.kafka.serializers.protobuf.AbstractKafkaProtobufDeserializer.deserializeWithSchemaAndVersion(AbstractKafkaProtobufDeserializer.java:294)
io.confluent.connect.protobuf.ProtobufConverter$Deserializer.deserialize(ProtobufConverter.java:200)\n\tat io.confluent.connect.protobuf.ProtobufConverter.toConnectData(ProtobufConverter.java:132)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.lambda$convertAndTransformRecord$4(WorkerSinkTask.java:536)\n\tat org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndRetry(RetryWithToleranceOperator.java:180)\n\tat org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:214)\n\tat org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execute(RetryWithToleranceOperator.java:156)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.convertAndTransformRecord(WorkerSinkTask.java:536)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.convertMessages(WorkerSinkTask.java:513)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:349)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:250)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:219)\n\tat org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:204)\n\tat org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:259)\n\tat org.apache.kafka.connect.runtime.isolation.Plugins.lambda$withClassLoader$1(Plugins.java:237)\n\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:840)\n
`

Can you tell me what could be wrong, probably I am doing something wrong.

**Comment by Paultagoras** [comment]
Hi @zTranquilizer do you happen to have an example of the data you're trying to send?

**Reply by Paultagoras** [reply]
That error suggests that you've set the Protobuf deserializer but it's not Protobuf data (so an exception is thrown) - was that what you intended to do?

**Reply by zTranquilizer** [reply, marked as answer]
> Hi @zTranquilizer do you happen to have an example of the data you're trying to send?

Unfortunately, I can't provide the data. I can prepare anonymous data later, if there is no solution without it.


> That error suggests that you've set the Protobuf deserializer but it's not Protobuf data (so an exception is thrown) - was that what you intended to do?

This is Protobuf data.
Found a feature, but there is a possibility that I am using something wrong.
I have everything working when I use image: confluentinc/cp-kafka-connect.
If I use image: confluentinc/cp-kafka-connect-base I get exactly the error I wrote about.

This is probably not your area of expertise anymore. But it would be great if you could tell me what could be wrong and why confluentinc/cp-kafka-connect works fine but confluentinc/cp-kafka-connect-base does not?
If there is no answer, I think my question will be closed.

**Reply by Paultagoras** [reply]
Sorry, not that familiar with the ins-and-outs of their docker images 😦
